app/controllers/books_controller.rb

```diff
@@ -3,13 +3,19 @@ class BooksController < ApplicationController
 
   # GET /books
   def index
-    @books = Book.all
+    if !params[:filter]
+      @books = Book.all  
```

Comment:
as a clean up you may think of making a separate def with books_query params. but i leave it up to you. this is ok for now.